scripts/cluster_network.py

```diff
@@ -153,9 +151,15 @@ def clustering_for_n_clusters(n, n_clusters, aggregate_carriers=None,
         aggregate_generators_carriers=aggregate_carriers,
         aggregate_one_ports=["Load", "StorageUnit"],
         line_length_factor=line_length_factor,
-        generator_strategies={'p_nom_max': p_nom_max_strategy}
-    )
-
+        generator_strategies={'p_nom_max': p_nom_max_strategy},
+        scale_link_capital_costs=False)
+
+    nc = clustering.network
+    nc.links['underwater_fraction'] = (n.links.eval('underwater_fraction * length')
+                                       .div(nc.links.length).dropna())
+    nc.links['capital_cost'] += (nc.links.length.sub(n.links.length).dropna()
+                                 .clip(lower=0)
+                                 .mul(extended_link_costs * line_length_factor))
```

Comment:
I think `line_length_factor` has already been considered to choose length and thus should not appear a second time for updating the `capital_cost`.